src/vgrid.f90

```diff
@@ -0,0 +1,210 @@
+    class(vgrid_t), intent(in)  :: vgrid_in
+
+    vgrid_out%type = vgrid_in%type
+
+    ! Dimension
+    vgrid_out%nlevels = vgrid_in%nlevels
+
+    ! Zeta variable and attributes
+    allocate(vgrid_out%zeta, source=vgrid_in%zeta)
+    if (allocated(vgrid_in%zeta_super)) then
+      allocate(vgrid_out%zeta_super, source=vgrid_in%zeta_super)
+    end if
+    if (allocated(vgrid_in%author)) then
+      vgrid_out%author = vgrid_in%author
+    end if
+
+    ! Global attributes
+    vgrid_out%original_file = vgrid_in%original_file
+    if (allocated(vgrid_in%history)) then
+      vgrid_out%history = vgrid_in%history
+    end if
+
+  end subroutine vgrid_copy
+
+  !-------------------------------------------------------------------------
+  subroutine vgrid_write(this, filename)
+    class(vgrid_t), target, intent(in) :: this
+    character(len=*), intent(in) :: filename
+
+    integer(int32) :: ncid, zeta_id, zeta_len, did(1) ! NetCDF ids
+    real(real64), pointer :: zeta_var(:)
+
+    write(output_unit,'(3a)') "Writing vgrid to file '", trim(filename), "'"
+
+    ! Handle the different types of grids
+    select case (this%type)
+    case ("mom5")
+      zeta_len = 2*this%nlevels + 1
+      zeta_var(1:zeta_len) => this%zeta_super(1:zeta_len)
+
+    case ("mom6")
+      zeta_len = this%nlevels + 1
+      zeta_var(1:zeta_len) => this%zeta(0:zeta_len - 1)
+    end select
+
+    ! Open file
+    call handle_error(nf90_create(trim(filename), ior(nf90_netcdf4, nf90_clobber), ncid))
+
+    ! Write dimension
+    call handle_error(nf90_def_dim(ncid, 'nzv', zeta_len, did(1)))
+
+    ! Write zeta
+    call handle_error(nf90_def_var(ncid, 'zeta', nf90_double, did, zeta_id))
+    call handle_error(nf90_put_att(ncid, zeta_id, 'units', 'meters'))
+    call handle_error(nf90_put_att(ncid, zeta_id, 'standard_name', 'vertical_grid_vertex'))
+    call handle_error(nf90_put_att(ncid, zeta_id, 'long_name', 'vgrid'))
+    call handle_error(nf90_put_att(ncid, zeta_id, 'author', trim(this%author)))
+    call handle_error(nf90_put_var(ncid, zeta_id, zeta_var))
+
+    ! Write global attributes
+    call handle_error(nf90_put_att(ncid, nf90_global, 'original_file', trim(this%original_file)))
+    call handle_error(nf90_put_att(ncid, nf90_global, 'history', trim(this%history)))
+
+    ! Close file
+    call handle_error(nf90_enddef(ncid))
+    call handle_error(nf90_close(ncid))
+
+  end subroutine vgrid_write
+
+  !-------------------------------------------------------------------------
+  subroutine vgrid_update_history(this, command)
+    use iso_c_binding
+    class(vgrid_t), intent(inout) :: this
+    character(len=*), intent(in) :: command
+
+    character(len=:), allocatable :: new_history, old_history
+
+    new_history = date_time() // ": " // trim(command)
+    if (allocated(this%history)) then
+      old_history = this%history
+      deallocate(this%history)
+      this%history = trim(new_history) // C_NEW_LINE // old_history
```

Comment:
reverse chronological order seems counterintuitive to me

Reply:
IIRC, this is how other tools that update the history do it, so I just kept it like that for consistency.